rbdmap build warning fixes for gcc/posix and MSVC/win32

SRSaunders opened the pull request:
This cleans up compiler warnings when building **rbdmap** on linux and win32:

1. Fixes rbdmap / idlib gcc PCH conflcts caused by a missing `USE_OPTICK` define.  Solution is to add the `USE_OPTICK` define to rbdmap PCH flags for non-MSVC builds.  Optick functionality is not needed, but the definition is required for gcc PCH consistency.  Tested on Manjaro linux.
2. Add specific gcc warning suppression for zlib and minizip third-party source libs.  Note this was already present for RBDoom3BFG, but was missing here.  Tested on Manjaro linux.
3. Replace deprecated `posix reader_r()` with `readdir()`, since it is no longer a thread safety issue. Note this was already done for RBDoom3BFG, again was missing here.  Tested on Manjaro linux.
4. Added specific MSVC warning suppression for imtui, imgui, pdcurses, and wincon third-party source libs.  Tested on Windows 10.

Note with these changes (plus adding a missing include) you can build and run rbdmap on macOS.  Not really needed at this time, and not enabled by default (I left your disable code alone), but it is there in case anyone needs it in the future.

RobertBeckebans commented:
Shouldn't then readdir_r() also replaced in Sys_ListFiles() in neo/sys/posix/posix_main.cpp ?

SRSaunders commented:
> Shouldn't then readdir_r() also replaced in Sys_ListFiles() in neo/sys/posix/posix_main.cpp ?

I replaced that one quite a while ago, along with the ones in `Sys_ReLaunch()` for platform_linux.cpp and platform_osx.cpp.  I just re-checked the code base and all instances of `readdir_r()` have been converted.
